tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/config/TmfJsonConfiguration.java

```diff
@@ -0,0 +1,294 @@
+     * Gets the parameters as JSON string
+     *
+     * @return the parameters as JSON string
+     */
+    @SuppressWarnings("null")
+    public String getJsonParameters() {
+        JsonElement parameters = fParameters;
+        if (parameters != null) {
+            return parameters.toString();
+        }
+        return "{}"; //$NON-NLS-1$
+    }
+
+    @Override
+    @SuppressWarnings("nls")
+    public String toString() {
+        return new StringBuilder(getClass().getSimpleName())
+            .append("[")
+            .append("fName=").append(getName())
+            .append(", fDescription=").append(getDescription())
+            .append(", fType=").append(getSourceTypeId())
+            .append(", fId=").append(getId())
+            .append(", fJsonParameters=").append(getJsonParameters())
+            .append("]").toString();
+    }
+
+    @Override
+    public boolean equals(@Nullable Object arg0) {
+        if (!(arg0 instanceof TmfJsonConfiguration)) {
+            return false;
+        }
+        TmfJsonConfiguration other = (TmfJsonConfiguration) arg0;
+        return Objects.equals(fName, other.fName) && Objects.equals(fId, other.fId)
+                && Objects.equals(fSourceTypeId, other.fSourceTypeId) && Objects.equals(fDescription, other.fDescription) && Objects.equals(fParameters, other.fParameters);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(fName, fId, fSourceTypeId, fDescription, fParameters);
+    }
+
+    private String toUuidString() {
+        String inputStr = new StringBuilder()
+                .append("fName=").append(fName) //$NON-NLS-1$
+                .append("fParameters=").append(getJsonParameters()).toString(); //$NON-NLS-1$
+        return UUID.nameUUIDFromBytes(Objects.requireNonNull(inputStr.getBytes(Charset.defaultCharset()))).toString();
+    }
+
+    /**
+     * Converts a JSON parameters string to {@link ITmfConfiguration}
+     *
+     * @param json
+     *            the parameters as JSON string
+     * @param defaultValues
+     *            {@link ITmfConfiguration} with default values for name,
+     *            description and sourceTypeId
+     * @return {@link ITmfConfiguration}
+     * @throws TmfConfigurationException
+     *             if an error occurred
+     * @since 9.5
+     */
+    public static ITmfConfiguration fromJsonString(String json, ITmfConfiguration defaultValues) throws TmfConfigurationException {
+       try {
+           @SuppressWarnings("null")
+           TmfJsonConfiguration config = new Gson().fromJson(json, TmfJsonConfiguration.class);
+           TmfConfiguration.Builder builder = new TmfConfiguration.Builder();
+           String name = config.getName();
+           name = name == null ? defaultValues.getName() : name;
+           String description = config.getDescription();
+           description = description == null ? defaultValues.getDescription() : description;
+           return builder.setId(config.getId())
```

Comment:
It seems a bit odd that we create a TmfConfiguration with id, name, description that can come from the json string and then the TmfConfiguration also contains the json string, so these values are duplicated inside the TmfConfiguration object.